src/ray/raylet_rpc_client/raylet_client_with_io_context.cc

```diff
@@ -31,8 +30,7 @@ RayletClientWithIoContext::RayletClientWithIoContext(const std::string &ip_addre
                                                      int port) {
   // Connect to the raylet on a singleton io service with a dedicated thread.
   // This is to avoid creating multiple threads for multiple clients in python.
-  static InstrumentedIOContextWithThread io_context("raylet_client_io_service");
-  instrumented_io_context &io_service = io_context.GetIoService();
+  instrumented_io_context &io_service = io_context_.GetIoService();
```

Comment:
Why changing this, seems unrelated to this PR

Reply:
Ah @sunsetxh is fixing your old comments https://github.com/ray-project/ray/pull/57004#discussion_r2449226163